### DISCUSSION TITLE
gyp installing dependencies

### DISCUSSION THREAD

**Discussion by Adem-Kri** [eclipse-theia/theia#12850] [Q&A]
Hello,
I've been trying to build my project but I keep facing the same error which I couldn't solve, I'm looking for any suggestions
thanks in advance

```bash
[4/4] Building fresh packages...
error C:\Users\\node_modules\drivelist: Command failed.
Exit code: 1
Command: prebuild-install || node-gyp rebuild
Arguments:
Directory: C:\Users\node_modules\drivelist
Output:
prebuild-install WARN install No prebuilt binaries found (target=16.18.1 runtime=node arch=x64 libc= platform=win32)
gyp info it worked if it ends with ok
gyp info using node-gyp@7.1.2
gyp info using node@16.18.1 | win32 | x64
gyp info find Python using Python version 3.10.8 found at "C:\Python310\python.exe"
gyp http GET https://nodejs.org/download/release/v16.18.1/SHASUMS256.txt
gyp http GET https://nodejs.org/download/release/v16.18.1/win-x86/node.lib
gyp http GET https://nodejs.org/download/release/v16.18.1/win-arm64/node.lib
gyp http GET https://nodejs.org/download/release/v16.18.1/win-x64/node.lib
gyp WARN install got an error, rolling back install
gyp ERR! configure error
gyp ERR! stack Error: tunneling socket could not be established, statusCode=407
gyp ERR! stack     at ClientRequest.onConnect (C:\Users\\node_modules\tunnel-agent\index.js:166:19)
gyp ERR! stack     at Object.onceWrapper (node:events:628:26)
gyp ERR! stack     at ClientRequest.emit (node:events:513:28)
gyp ERR! stack     at Socket.socketOnData (node:_http_client:574:11)
gyp ERR! stack     at Socket.emit (node:events:513:28)
gyp ERR! stack     at addChunk (node:internal/streams/readable:315:12)
gyp ERR! stack     at readableAddChunk (node:internal/streams/readable:289:9)
gyp ERR! stack     at Socket.Readable.push (node:internal/streams/readable:228:10)
gyp ERR! stack     at TCP.onStreamRead (node:internal/stream_base_commons:190:23)
gyp ERR! System Windows_NT 10.0.22621
gyp ERR! command "C:\\Users\\tools\\node-16.18.1\\node.exe" "C:\\Users\\\\node_modules\\@npmcli\\run-script\\node_modules\\node-gyp\\bin\\node-gyp.js" "rebuild"
gyp ERR! cwd C:\Users\\node_modules\drivelist
gyp ERR! node -v v16.18.1
gyp ERR! node-gyp -v v7.1.2
gyp ERR! not ok
info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command.
```
![image](https://github.com/eclipse-theia/theia/assets/116633370/b9afdf70-e5cc-43d9-8e1f-e69fe65373dc)

**Comment by msujew** [comment]
@Adem-Kri Looks like the version of `node-gyp` you're using is quite outdated. `7.1.2` is more than 3 years old. Theia expects you to use `^9.0.0`, see [here](https://github.com/eclipse-theia/theia/blob/master/package.json#L45C18-L45C24).

**Reply by Adem-Kri** [reply]
@msujew Even after upgrading the node-gyp to `9.4.0` it is yet showing the same error message  

![image](https://github.com/eclipse-theia/theia/assets/116633370/0206a4a5-7f43-4d44-815f-148f2d1b18b5)

**Reply by msujew** [reply]
That seems to be a different error message than the previous one...

Anyway, you should probably clean your node_modules folder. I assume that the old node-gyp version broke something in your dependencies. If it still does not work, please provide a reproducible example so that we can look into the issue more closely.

**Reply by Adem-Kri** [reply]
@msujew 
After deleting the node_modules it still the same

![image](https://github.com/eclipse-theia/theia/assets/116633370/8f0b0f8e-d5f5-4674-ae56-93ff5fff20bd)

**Reply by msujew** [reply]
@Adem-Kri That's also a different error than the previous two issues.

Are you able to visit the https://nodejs.org/download/release page? Looking at the error (http 407), it seems like your proxy is blocking access to the page. You might need to set the `HTTP_PROXY`/`HTTPS_PROXY` env variables for the download to succeed.
